iphone/Classes/TiUIListView.m

```diff
@@ -1492,6 +1505,65 @@ - (NSString *)tableView:(UITableView *)tableView titleForFooterInSection:(NSInte
     return [[self.listViewProxy sectionForIndex:section] footerTitle];
 }
 
+#pragma mark - UITableViewDataSourcePrefetching
+
+#if IS_XCODE_8
+- (void)tableView:(UITableView *)tableView prefetchRowsAtIndexPaths:(NSArray<NSIndexPath *> *)indexPaths
+{
+    NSString *eventName = @"prefetch";
+    if (![self.proxy _hasListeners:eventName]) {
+        return;
+    }
+    
+    NSMutableArray *cells = [[NSMutableArray arrayWithCapacity:[indexPaths count]] retain];
+    
+    for (NSIndexPath *indexPath in indexPaths) {
+        [cells addObject:[self listItemFromIndexPath:indexPath]];
+    }
+
+    [self.proxy fireEvent:eventName withObject:@{@"prefetchedItems": cells}];
+    RELEASE_TO_NIL(cells);
+}
+
+- (void)tableView:(UITableView *)tableView cancelPrefetchingForRowsAtIndexPaths:(NSArray<NSIndexPath *> *)indexPaths
+{
+    NSString *eventName = @"cancelprefetch";
+    if (![self.proxy _hasListeners:eventName]) {
+        return;
+    }
+    
+    NSMutableArray *cells = [[NSMutableArray arrayWithCapacity:[indexPaths count]] retain];
+    
+    for (NSIndexPath *indexPath in indexPaths) {
+        [cells addObject:[self listItemFromIndexPath:indexPath]];
+    }
+    
+    [self.proxy fireEvent:eventName withObject:@{@"prefetchedItems": cells}];
+    RELEASE_TO_NIL(cells);
+}
+#endif
+
+- (NSDictionary*)listItemFromIndexPath:(NSIndexPath*)indexPath
+{
+    NSIndexPath *realIndexPath = [self pathForSearchPath:indexPath];
+    
+    TiUIListSectionProxy *section = [self.listViewProxy sectionForIndex:realIndexPath.section];
+    NSDictionary *item = [section itemAtIndex:realIndexPath.row];
+    NSMutableDictionary *eventObject = [[NSMutableDictionary alloc] initWithObjectsAndKeys:
```

Comment:
doing alloc here means we have to release it eventually. maybe add `autorelease`.
OR, do `dictionaryWithCapacity` instead, then set each key and object.